tests/unit/libbpf_test.cpp

```diff
@@ -3771,3 +3771,68 @@ _utility_test(ebpf_execution_type_t execution_type)
 }
 
 DECLARE_ALL_TEST_CASES("utility_test", "[libbf]", _utility_test);
+
+static void
+_program_flags_test(ebpf_execution_type_t execution_type)
+{
+    _test_helper_end_to_end test_helper;
+    const char dll_name[] = "utility_um.dll";
+    const char obj_name[] = "utility.o";
+    test_helper.initialize();
+    single_instance_hook_t hook(EBPF_PROGRAM_TYPE_BIND, EBPF_ATTACH_TYPE_BIND);
+    REQUIRE(hook.initialize() == EBPF_SUCCESS);
+    program_info_provider_t sample_program_info;
+    REQUIRE(sample_program_info.initialize(EBPF_PROGRAM_TYPE_BIND) == EBPF_SUCCESS);
+
+    const char* file_name = (execution_type == EBPF_EXECUTION_NATIVE ? dll_name : obj_name);
+    struct bpf_object* process_object = bpf_object__open(file_name);
+    REQUIRE(process_object != nullptr);
+
+    // Set the flag on each program in the object.
+    struct bpf_program* program;
+    bpf_object__for_each_program(program, process_object)
+    {
+        // Set some flags on the program. The test attach provider does not use these flags.
+        REQUIRE(bpf_program__set_flags(program, 0xCCCCCCCC) == 0);
+
+        // Get the flags and verify they are correct.
```

Comment:
All this does is verify that the flags are retained while the program object is still in user-mode memory.  A stricter test might pin the program, release this reference so all user mode state is freed, and then recreate the object from the pin path and see if the flags were kept by the kernel.

Reply:
The test also queries the flags from the attach provider. This confirms they were set in kernel mode and correctly transmitted over the attach NMR contract. Seel line 3819

Reply:
But I don't see that it tests that the user mode API can correctly get the data from kernel mode.  If it does, where is that tested?

Reply:
Is that behavior supported on Linux? From what I can tell from the libbpf code, it's an input that is passed when programs are loaded into kernel, but I don't see a mechanism to query it from kernel.

Reply:
No idea, since the API to get flags isn't documented.

Reply:
From looking at the libbpf code and linux/bpf.h I think Linux doesn't allow it (in any way I can find).  It would be possible to add flags to `bpf_prog_info` in include\ebpf_structs.h if there is a need and that would allow retrieval via `bpf_obj_get_info_by_fd()`.